plugins/balloonpanel/plugin.js

```diff
@@ -734,6 +825,16 @@
 					left: elementRect.right - elementRect.width / 2 - panelWidth + this.triangleMinDistance
 				}
 			};
+
+			if ( requestedAlignments ) {
```

Comment:
I'd think about conditionally calculating alignments than deleting the unnecessary ones.

OTOH is it really so crucial to return only the subset of alignments? Can't the unnecessary ones be just ignored?